File: FlaskServer/domain/InterestedPerson.py
from sqlalchemy import Column, Integer, String, Date, ForeignKey
from sqlalchemy.orm import relationship
from infra.AppDataBase import Base


class InterestedPerson(Base):
    __tablename__ = 'interested_person'
    id = Column(Integer, primary_key=True)
    reference_id = Column(String, nullable=False)
    name = Column(String, nullable=False)
    phone_number = Column(String, nullable=False)
    formatted_phone_number = Column(String)
    course_ordered = Column(String)
    order_date = Column(Date)
    address = Column(String)
    status = Column(String)
    last_status_update = Column(Date)

    responsible_id = Column(Integer, ForeignKey('responsible.id'))
    responsible = relationship("Responsible", back_populates="interested_people")

    def get_dict(self):
        return {
            'id': self.id,
            'name': self.name,
            'address': self.address,
            'phoneNumber': self.phone_number,
            'formattedPhoneNumber': self.formatted_phone_number,
            'courseOrdered': self.course_ordered,
            'orderDate': self.order_date,
            'status': self.status,
            'responsible': self.responsible.name
        }

File: evangelism-assistant-server/src/main/kotlin/com/quenazapps/evangelism_assistant/domain/InterestedPerson.kt
package com.quenazapps.evangelism_assistant.domain

import com.quenazapps.evangelism_assistant.domain.DTO.InterestedPersonDTO
import jakarta.persistence.CascadeType
import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.Table
import java.time.LocalDate

@Entity
@Table(name = "interested_person")
data class InterestedPerson(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Int = 0,

    @Column(nullable = false)
    val referenceId: String,

    @Column(nullable = false)
    val name: String,

    @Column(nullable = false)
    val phoneNumber: String,

    val formattedPhoneNumber: String? = null,

    val courseOrdered: String? = null,

    val orderDate: LocalDate? = null,

    val address: String? = null,

    val status: String? = null,

    val lastStatusUpdate: LocalDate? = null,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "responsible_id")
    val responsible: Responsible? = null
){
    fun toDTO(): InterestedPersonDTO {
        return InterestedPersonDTO(
            id = this.id,
            name = this.name,
            phoneNumber = this.phoneNumber,
            formattedPhoneNumber = this.formattedPhoneNumber,
            status = this.status,
            referenceId = this.referenceId,
            courseOrdered = this.courseOrdered,
            orderDate = this.orderDate,
            address = this.address,
            lastStatusUpdate = this.lastStatusUpdate,
            responsible = this.responsible?.name,
        )
    }
}